src/librustc_mir/build/matches/simplify.rs

```diff
@@ -106,27 +106,28 @@ impl<'a, 'gcx, 'tcx> Builder<'a, 'gcx, 'tcx> {
             }
 
             PatternKind::Range(PatternRange { lo, hi, ty, end }) => {
```

Comment:
Since this code apparently wasn't obvious to even yourself, a few comments would be nice ;)